### PR TITLE
Add new config.jwtDomain value to oidc-discovery chart

### PR DESCRIPTION
**Pull request by erikgb**
This adds a new value to the spiffe-oidc-discovery-provider chart that avoids Helm trying to `urlParse` the `jwtIssuer` value. This should resolve https://github.com/spiffe/helm-charts-hardened/issues/455 in a non-breaking way.

````
$ helm template charts/spire --set 'global.spire.jwtIssuer=https://spire.apps.${CLUSTER_DOMAIN}' --set 'spiffe-oidc-discovery-provider.config.jwtDomain=spire.apps.${CLUSTER_DOMAIN}' | grep jwt_issuer
        "jwt_issuer": "https://spire.apps.${CLUSTER_DOMAIN}",
````

Fixes https://github.com/spiffe/helm-charts-hardened/issues/455

### CODE REVIEW

**Comment by erikgb** [comment]
The failing tests seem unrelated to the proposed changes, but please correct me if I'm wrong.

**Comment by kfox1111** [comment]
There were some network issues on the last run. I reran things.

This one looks like it is related to the pr:
```
Name:             spire-spiffe-oidc-discovery-provider-test-connection
...
  curl-ingress:
    Container ID:  containerd://d27394f6cc63456811d7649824fef41c40b468c11d0f3f397325bb7af09a68a7
    Image:         cgr.dev/chainguard/bash:latest@sha256:3ed6a0b7b498b531c631a5f314ba85e91c6fd66d47301f9f67e9422e395c80c8
    Image ID:      cgr.dev/chainguard/bash@sha256:3ed6a0b7b498b531c631a5f314ba85e91c6fd66d47301f9f67e9422e395c80c8
    Port:          <none>
    Host Port:     <none>
    Command:
      curl
    Args:
      -s
      -f
      --cacert
      /ca/ca.crt
      https://oidc-discovery.production.other/.well-known/openid-configuration
    State:          Terminated
      Reason:       Error
      Exit Code:    22
      Started:      Sat, 14 Sep 2024 02:08:20 +0000
      Finished:     Sat, 14 Sep 2024 02:08:20 +0000
    Ready:          False
    Restart Count:  0
    Environment:    <none>
    Mounts:
      /ca from custom-ca (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-wz76c (ro)
```

**Comment by kfox1111** [comment]
Yeah... the default rendering is broken:
```
      "domains": [
        "release-name-spiffe-oidc-discovery-provider",
        "release-name-spiffe-oidc-discovery-provider.default",
        "release-name-spiffe-oidc-discovery-provider.default.svc.cluster.local",
        ""
      ],
```

Will make a suggestion to fix it.

File: charts/spire/charts/spiffe-oidc-discovery-provider/templates/configmap.yaml [hunk 1]
```diff
@@ -23,9 +23,12 @@ domains:
   - "{{ include "spiffe-oidc-discovery-provider.fullname" . }}"
   - "{{ include "spiffe-oidc-discovery-provider.fullname" . }}.{{ include "spiffe-oidc-discovery-provider.namespace" . }}"
   - "{{ include "spiffe-oidc-discovery-provider.fullname" . }}.{{ include "spiffe-oidc-discovery-provider.namespace" . }}.svc.{{ include "spire-lib.cluster-domain" . }}"
-  {{- $uri := urlParse (include "spire-lib.jwt-issuer" .) }}
-  {{- $jwtIssuer := (default $uri.path $uri.host) }}
-  {{- uniq (concat (list $jwtIssuer) .Values.config.additionalDomains) | toYaml | nindent 2 }}
+  {{- $jwtDomain := .Values.config.jwtDomain }}
+  {{- if not $jwtDomain }}
+    {{- $uri := urlParse (include "spire-lib.jwt-issuer" .) }}
+    {{- $jwtDomain := (default $uri.path $uri.host) }}
```
Comment (kfox1111) [inline charts/spire/charts/spiffe-oidc-discovery-provider/templates/configmap.yaml:29]:
```suggestion
  {{- if eq $jwtDomain "" }}
    {{- $uri := urlParse (include "spire-lib.jwt-issuer" .) }}
    {{- $jwtDomain = (default $uri.path $uri.host) }}
```

Comment (erikgb) [reply charts/spire/charts/spiffe-oidc-discovery-provider/templates/configmap.yaml]: Empty string is "falsy", ref. https://helm.sh/docs/chart_template_guide/control_structures/#ifelse. Thanks for the var fix suggestion! It's crazy that Helm (or Go-template) allows me to redefine/shadow an existing variable. 😅

**Review by kfox1111 — APPROVED** [review]
LGTM. Thanks! :)

**Pull request merged**